Improve linear model (#2451)
openturns/openturns: python/doc/theory/meta_modeling/polynomial_least_squares.rst

         \sum_{i=1}^{n_{X}} \sum_{j=1}^{n_{X}} a_{i,j} x_i x_j

The two previous equations may be recast as:

  .. math::

      y  \approx  \widehat{\model}(\vect{x})  =  \sum_{j=0}^{P-1} a_j \psi_j(\vect{x})

where :math:`P` denotes the number of terms, which is equal to
:math:`(n_X + 1)` (resp. to :math:`(1 + 2n_X + n_X (n_X - 1)/2)`) when
using a linear (resp. a quadratic) approximation, and the family
:math:`(\psi_j,j=0,\dots,P-1)` gathers the constant monomial
:math:`1`, the monomials of degree one :math:`x_i` and possibly the
cross-terms :math:`x_i x_j` as well as the monomials of degree two
:math:`x_i^2`. Using the vector notation
:math:`\vect{a}  =  (a_{0} , \dots , a_{P-1} )^{\textsf{T}}`
and
:math:`\vect{\psi}(\vect{x})  =  (\psi_{0}(\vect{x}) , \dots , \psi_{P-1}(\vect{x}) )^{\textsf{T}}`,
this rewrites:

  .. math::

      y  \approx  \widehat{\model}(\vect{x})  =  \vect{a}^{\textsf{T}} \vect{\psi}(\vect{x})

A *global* approximation of the model response over its whole
definition domain is sought. To this end, the coefficients :math:`a_j`
may be computed using a least squares regression approach. In this

> context, an experimental design that is, a set of observations of

Reviewer: ```suggestion
context, an experimental design, that is, a set of observations of
```

input parameters, is required:

.. math::

    \cX = \left\{ \vect{x}^{(1)}, \dots, \vect{x}^{(N)} \right\},

as well as the corresponding model evaluations:

.. math::

    \cY = \left\{ y^{(1)},\dots,y^{(N)} \right\}.

The following minimization problem has to be solved:

.. math::

    \mbox{Find} \quad \widehat{\vect{a}} \quad \mbox{that minimizes} \quad \cJ(\vect{a})  =  \sum_{i=1}^N \left( y^{(i)} - \vect{a}^{\textsf{T}}  \vect{\psi}\left(\vect{x}^{(i)}\right) \right)^2

Let :math:`\vect{y} = (y^{(1)},\dots,y^{(N)})^T \in \Rset^{N}` be the
vector of output observations.
The solution is given by the normal equations:

.. math::

>     \widehat{\vect{a}}  =  \left( \boldsymbol{\Psi}^{\textsf{T}} \boldsymbol{\Psi}  \right)^{-1} \boldsymbol{\Psi}^{\textsf{T}}  \vect{y}

Reviewer: Here and elsewhere `boldsymbol` should be replaced with `mat`

Author: You are right. Even more: this is a matrix and the `\mat` macro from maths_notations.sty is for that specific purpose. By the way, the `{\textsf{T}` can be replaced with the `\Tr` macro. 

Fixed at 2d79cecad.

where:

.. math::

    \boldsymbol{\Psi}  =  (\psi_{j}\left(\vect{x}^{(i)}\right) , i=1,\dots,N , j = 0,\dots,P-1)

It is clear that the above equation is only valid for a full rank

> information matrix. A necessary condition is that the size :math:`N`

Reviewer: ```suggestion
information matrix :math:`\mat{\Psi}`. A necessary condition is that the size :math:`N`
```

Author: No: in a regression setting, the information matrix is $\frac{1}{\sigma^2} \Psi^T \Psi$, where $\Psi$ is the design matrix (see [math.stackexchange.com](https://math.stackexchange.com/questions/328835/get-a-fisher-information-matrix-for-linear-model-with-the-normal-distribution-fo)). Equivalently, the covariance matrix is $\sigma^2 (\Psi^T \Psi)^{-1}$. Do you agree on this point?

Reviewer: Indeed, you are right! But I think this should be explicitly stated.

Author: You suggest to add a content which is related to the linear regression setting, which is different from the least squares approximation setting presented in this page. This is why I think that it is better not to introduce the information matrix here. I send you an email for a more detailed analysis of this issue. Please tell me if that fits your analysis.

Reviewer: But it is strange to mention a matrix without being able to write it! Maybe remove the sentence entirely then?

Author: Ok, I got it. You are right, the text was confusing: this is the design matrix that needs to be full rank, not the information matrix.

of the experimental design is not less than the number :math:`P` of
coefficients to estimate. In practice, it is not recommended to
directly invert
:math:`\boldsymbol{\Psi}^{\textsf{T}} \boldsymbol{\Psi}`
since the solution may be particularly sensitive to an

> ill-conditioning of the matrix. The least-square problem is rather

Reviewer: ```suggestion
ill-conditioned matrix. The least-square problem is rather
```
